torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -785,6 +792,14 @@ void Reducer::initialize_buckets(
         at::TensorOptions options;
         size_t offset = 0;
 
+        // Reserve enough space for the per-variable fields stored in bucket
+        // replica for efficiency.
+        num_variables += bucket_indices[bucket_index].size();
+        replica.variables.reserve(num_variables);
+        replica.offsets.reserve(num_variables);
+        replica.lengths.reserve(num_variables);
+        replica.sizes_vec.reserve(num_variables);
+
```

Comment:
Not completely sure I follow this logic. Don't we just want to reserve `bucket_indices[bucket_index].size()` for each replica? Why are we incrementing `num_variables` and using that as the argument into `reserve`?

Reply:
You are right. I shouldn't increment `num_variables ` here. This wrong logic was not captured before because there is only one replica in this mode. Thanks!